imapautofiler/client.py

```diff
@@ -190,7 +191,7 @@ def list_mailboxes(self):
 
     def mailbox_iterate(self, mailbox_name):
         self._conn.select_folder(mailbox_name)
-        msg_ids = self._conn.search(['ALL'])
+        msg_ids = self._conn.search(self.search)
```

Comment:
Is looks like this passes a string, where the old code passed a list with a string in it. I don't remember enough of the client API to remember if those are equivalent. Did you mean to wrap this in a list, or is passing the string effectively the same?

Reply:
We should also think about whether this behavior can be reproduced for the MaildirClient, and if not log a warning at least that the parameter is being ignored in that case.

Reply:
> Is looks like this passes a string, where the old code passed a list with a string in it. I don't remember enough of the client API to remember if those are equivalent. Did you mean to wrap this in a list, or is passing the string effectively the same?

It's the same : https://github.com/mjs/imapclient/blob/391cc6d66d35c16bed11292ff00835646cd50ae5/imapclient/imapclient.py#L1832-L1833
String/bytes are convert everytime

For the MaildirClient, I think a warning should be good

Reply:
good for you ?